lib/OpenQA/Worker/Job.pm

```diff
@@ -653,37 +655,23 @@ sub is_backend_running {
     return !exists $engine->{child} ? !!0 : $engine->{child}->is_running;
 }
 
-sub start_livelog {
-    my ($self) = @_;
-
+# starts/stops the livelog
+sub start_livelog ($self) { $self->_set_livelog_viewers($self->livelog_viewers + 1) }
+sub stop_livelog ($self) { $self->_set_livelog_viewers(max(0, $self->livelog_viewers - 1)) }
+sub _set_livelog_viewers ($self, $livelog_viewers = undef) {
+    $self->{_livelog_viewers} = $livelog_viewers if defined $livelog_viewers;
     return undef unless $self->is_backend_running;
-
     my $pooldir = $self->worker->pool_directory;
-    my $livelog_viewers = $self->livelog_viewers + 1;
-    if ($livelog_viewers == 1) {
+    if ((!defined $livelog_viewers && $self->{_livelog_viewers} >= 1) || ($self->{_livelog_viewers} == 1)) {
```

Comment:
I'm not sure I get what the semantics of an undefined number of viewers are. It's not starting or stopping... but it's somehow used when there's viewers before the live log is enabled? I would suggest to make whatever it does more explicit.

Reply:
Maybe I should add a comment about this. The `$livelog_viewers` variable is suppose to be `undef` if the current number of viewers should be re-evaluated. This is done once the backend is running.

Reply:
This code was even buggy when the number of viewers dropped from 2 to 1 (in which it would try to start the live log but that's not necessary). So I changed this function quite a bit. Now these conditions should make much more sense.